apps/bmd/src/AppRoot.tsx

```diff
@@ -141,898 +147,1091 @@ export const activationStorageKey = 'activation'
-      this.props.storage.set(activationStorageKey, data)
+    const contestIndex = election.contests.findIndex((c) => c.id === contestId)
+    /* istanbul ignore next */
+    if (contestIndex < 0) {
+      throw new Error(`No contest found for contestId: ${contestId}`)
     }
-  }
-
-  public retrieveVotes = (): VotesDict =>
-    this.props.storage.get(votesStorageKey) || blankBallotVotes
-
-  public storeVotes = async (votes: VotesDict): Promise<void> => {
+    const contestTally = tally[contestIndex]
+    const contest = election.contests[contestIndex]
     /* istanbul ignore else */
-    if (process.env.NODE_ENV !== 'production') {
-      this.props.storage.set(votesStorageKey, votes)
+    if (contest.type === 'yesno') {
+      const yesnoContestTally = contestTally as YesNoVoteTally
+      const vote = votes[contestId] as YesNoVote
+      yesnoContestTally[getSingleYesNoVote(vote)!]++
+    } else if (contest.type === 'candidate') {
+      const candidateContestTally = contestTally as CandidateVoteTally
+      const vote = votes[contestId] as CandidateVote
+      vote.forEach((candidate) => {
+        if (candidate.isWriteIn) {
+          const tallyContestWriteIns = candidateContestTally.writeIns
+          const writeIn = tallyContestWriteIns.find(
+            (c) => c.name === candidate.name
+          )
+          if (typeof writeIn === 'undefined') {
+            tallyContestWriteIns.push({
+              name: candidate.name,
+              tally: 1,
+            })
+          } else {
+            writeIn.tally++
+          }
+        } else {
+          const candidateIndex = contest.candidates.findIndex(
+            (c) => c.id === candidate.id
+          )
+          if (
+            candidateIndex < 0 ||
+            candidateIndex >= candidateContestTally.candidates.length
+          ) {
+            throw new Error(
+              `unable to find a candidate with id: ${candidate.id}`
+            )
+          }
+          candidateContestTally.candidates[candidateIndex]++
+        }
+      })
     }
-
-    this.lastVoteUpdateAt = Date.now()
-  }
-
-  public resetVoterData = (): void => {
-    this.props.storage.remove(activationStorageKey)
-    this.props.storage.remove(votesStorageKey)
-  }
-
-  public unconfigure = (): void => {
-    this.setState(this.initialUserState)
-    this.props.storage.clear()
-    this.props.history.push('/')
-  }
-
-  public storeAppState = (): void => {
-    const {
-      appPrecinctId,
-      ballotsPrintedCount,
-      isLiveMode,
-      isPollsOpen,
-      tally,
-    } = this.state
-    this.props.storage.set(stateStorageKey, {
-      appPrecinctId,
-      ballotsPrintedCount,
-      isLiveMode,
-      isPollsOpen,
-      tally,
-    })
   }
+  return tally
+}
 
-  public retrieveAppState = (): Partial<State> => {
-    return this.props.storage.get(stateStorageKey) || {}
+// Sets State. All side effects done outside: storage, fetching, etc
+type AppActions =
```

Comment:
Seems like this should be singular, not plural. Unless there's a reason I'm missing?